src/d/actor/d_a_fganon.cpp

```diff
@@ -5,195 +5,2885 @@
+        local_7c.x = 20000.0f;
+        local_7c.y = 20000.0f;
+        local_7c.z = z;
+    }
+
+    i_this->mCyl.SetC(local_7c);
+    i_this->mCyl.SetH(REG8_F(3) + 250.0f);
+    i_this->mCyl.SetR(REG8_F(4) + 70.0f);
+
+    dComIfG_Ccsp()->Set(&i_this->mCyl);
+    
+    local_84.x = -10000.0f;
+    local_84.y = -10000.0f;
+    local_84.z = -10000.0f;
+
+    if (i_this->m685 != 0) {
+        i_this->mWeponSph.SetR(REG0_F(13) + 60.0f);
+        mDoMtx_copy(i_this->mpMorf->getModel()->getAnmMtx(24), *calc_mtx);
+        local_70.x = 0.0f;
+        local_70.y = 0.0f;
+        local_70.z = 100.0f;
+        MtxPosition(&local_70, &local_84);
+        i_this->m685 = 0;
+        if (i_this->m684 == 0) {
+            i_this->mWeponSph.StartCAt(local_84);
+            i_this->m684 = 1;
+        }
+        else {
+            i_this->mWeponSph.MoveCAt(local_84);
+        }
+    } else {
+        i_this->m684 = 0;
+        i_this->mWeponSph.SetC(local_84);
+    }
+    g_dComIfG_gameInfo.play.mCcS.Set(&i_this->mWeponSph);
+    i_this->m686 = 0;
+    mant_class* cape = (mant_class*)fopAcM_SearchByID(i_this->mCapeID);
+    if (cape != NULL) {
+        mDoMtx_copy(i_this->mpMorf->mpModel->mpNodeMtx[REG0_S(5) + 20], *calc_mtx);
+        local_70.x = REG0_F(0) + 35.0f;
+        local_70.y = REG0_F(1);
+        local_70.z = REG0_F(2) + -30.0f;
+        MtxPosition(&local_70, &cape->m1BE0);
+        mDoMtx_copy(i_this->mpMorf->mpModel->mpNodeMtx[REG0_S(6) + 11], *calc_mtx);
+        local_70.x = REG0_F(3) + 35.0f;
+        local_70.y = REG0_F(4);
+        local_70.z = REG0_F(5) + 30.0f;
+        MtxPosition(&local_70, &cape->m1BEC);
+        cape->current.pos = i_this->eyePos;
+        cape->current.angle = i_this->current.angle;
+        float fVar1;
+        if ((i_this->mbIsMaterialized != 0) && (i_this->m68E == 0)) {
+            fVar1 = 1.0f;
+        }
+        else {
+            fVar1 = 0.0f;
+        }
+        cLib_addCalc2(&cape->scale.y, fVar1 , 1.0f, REG0_F(8) + 0.1f);
+    }
+    if (i_this->m6AC != 0) {
+        i_this->m6AC--;
+        i_this->setBtAttackData(0.0f, 10.0f, 10000.0f, 1);
+        i_this->setBtMaxDis(l_HIO.m44);
+        i_this->setBtNowFrame(5.0f);
+    }
+    else {
+        i_this->setBtNowFrame(100.0f);
+    }
+    return TRUE;
 }
 
 /* 0000924C-00009254       .text daFganon_IsDelete__FP12fganon_class */
 static BOOL daFganon_IsDelete(fganon_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00009254-000093B4       .text daFganon_Delete__FP12fganon_class */
-static BOOL daFganon_Delete(fganon_class*) {
-    /* Nonmatching */
+static BOOL daFganon_Delete(fganon_class* i_this) {    
+    if ((i_this->mB8A) && (dComIfG_resDelete(&i_this->mPhs2, "Pgsw"), i_this->mB8A != 1))
+        dComIfG_resDelete(&i_this->mPhs1, "Fganon");
+    
+    if (i_this->mB88 != 0) {
+        hio_set = 0;
+        mDoHIO_deleteChild(l_HIO.mNo);
+    }
+
+    mDoAud_seDeleteObject(&i_this->m3E0);
+
+    if (i_this->mCapeID != 0xFFFF)
```

Comment:
It's weird that they'd write it like this since since the invalid value for procids should be 0xFFFFFFFF (`fpcM_ERROR_PROCESS_ID_e`), not 0xFFFF. I'd add a comment like this so someone knows to look into it later, maybe this causes some kind of bug in the game somewhere.
```suggestion
    if (i_this->mCapeID != 0xFFFF) // bug?
```